otp_provisioning/generate_factory_bin_product.py

```diff
@@ -0,0 +1,149 @@
+#
+# 1. Redistributions of source code must retain the above copyright notice,
+# this list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following disclaimer in the documentation
+# and/or other materials provided with the distribution.
+#
+# 3. Neither the name of the copyright holder nor the names of its contributors
+# may be used to endorse or promote products derived from this software without
+# specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import sys
+import argparse
+import ctypes
+import binascii
+
+import factory_data
+
+base_addr = hex(0xffffe000 + 0x200)
+
+GENUINO = "Genuino 101"
+ARDUINO = "Arduino 101"
+VENDOR = "Intel"
+# version of this OTP block, so that it can be extended or changed in the future.
+VERSIONHI = 2
+VERSIONLO = 0
+KEYSTART = 0xA5A5A5A5
+KEYEND = 0x5A5A5A5A
+
+def main():
+# Parse input arguments
+    parser = argparse.ArgumentParser(description="Generate a binary file containing"
+        "various provisioning product info such as product serial number.")
+    parser.add_argument('--product_sn', help="the product serial number, up to 16 ASCII characters", required=True)
+    parser.add_argument('--product_hw_ver', help="the product hardware version, up to 4 bytes in hexadecimal", required=True)
+    parser.add_argument('--product_board_name', help="the product board name", required=True)
+    parser.add_argument('--product_vendor_name', help="the product vendor name", required=True)
+    parser.add_argument('--product_VID', help="the product VID", required=True)
+    parser.add_argument('--product_PID', help="the product PID", required=True)
+    args = parser.parse_args()
+
+
+# Create instance of customer data struct
+    product_data = factory_data.CustomerData()
+
+    # Add the product serial number
+    ############
+    assert len(args.product_sn) <= 16
+
+    product_sn_padded = args.product_sn.ljust(16, '\0')
+    product_data.product_sn = (ctypes.c_ubyte * 16).from_buffer_copy(bytearray(product_sn_padded))
+
+    print "Product S/N     : " + product_sn_padded
```

Comment:
These seem like debug prints.... all the printing of the final padded values. Did you mean to remove them?